### PR TITLE
Innitial work on PNLS report indicators

### PR DESCRIPTION
**Pull request by mozzy11**
New PR ,Agregated the data using indicators and dimensions rather than SQL queries

### CODE REVIEW

**Comment by mozzy11** [comment]
cc @moshonk  @ckemar.  Final PR on the first indicator.
All aggreagates can be disaggregated.

File: api/src/main/resources/messages.properties [hunk 1]
```diff
@@ -245,4 +245,29 @@ isanteplusreports.number_of_suspected_cases_of_malaria_tested=Number of suspecte
...
+isanteplusreports.pnls.result.20-24.label=20-24 Yrs
+isanteplusreports.pnls.result.25-29.label=25-29 Yrs
+isanteplusreports.pnls.result.30-34.label=30-34 Yrs
+isanteplusreports.pnls.result.35-39.label=35-39 Yrs
+isanteplusreports.pnls.result.40-44.label=40-44 Yrs
+isanteplusreports.pnls.result.45-49.label=45-49 Yrs
+isanteplusreports.pnls.result.50.label=50+ Yrs
+isanteplusreports.pnls.result.UnKnown.label=UnKnown Yrs
+isanteplusreports.pnls.result.Total.label=Total
+isanteplusreports.newlyenrolledPatentsonArt=Newly Enrolled Patients on ART in the Last Period
```
Comment (moshonk) [inline api/src/main/resources/messages.properties:270]: Please change this label to *NEW 1.1: Number of HIV+ newly enrolled on ART during the selected period (TX_NEW)*

Comment (moshonk) [reply api/src/main/resources/messages.properties:270]: Trivial change: 
Correct typo *isanteplusreports.newlyenrolledPatentsonArt* to *isanteplusreports.newlyEnrolledPatientsOnArt*

Comment (mozzy11) [reply api/src/main/resources/messages.properties]: corrected it . 
I added a second indicator . All works fine

File: api/src/main/java/org/openmrs/module/isanteplusreports/pnlsReport/utils/pnlsReportUtils.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+		cd.setDescription(MessageUtil.translate(name));
+		for (Parameter parameter : parameters) {
+			cd.addParameter(parameter);
+		}
+		
+		cd.setQuery(sql);
+		return cd;
+	}
+	
+	public static void addAgeandGenderColumns(CohortIndicatorDataSetDefinition dsd, CohortIndicator CohortIndicator,
```
Comment (moshonk) [inline api/src/main/java/org/openmrs/module/isanteplusreports/pnlsReport/utils/pnlsReportUtils.java:52]: Proper camel-casing
addAgeAndGenderColumns

File: omod/src/main/java/org/openmrs/module/isanteplusreports/conventer/StringToPnlsIndicatorListConverter.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+		List<PnlsReportSelectedIndicator> list = null;
+		try {
+			list = mapper.readValue(json, type);
+		}
+		catch (IOException ex) {
+			LOGGER.error("Cannot convert json to List<PnlsReportSelectedIndicator>>. Null was returned");
+		}
+		return list;
+	}
+}
```
Comment (moshonk) [inline omod/src/main/java/org/openmrs/module/isanteplusreports/conventer/StringToPnlsIndicatorListConverter.java:32]: This class is not referenced anywhere. Please remove it.

File: api/src/main/resources/messages_fr.properties [hunk 4]
```diff
@@ -236,3 +236,31 @@ isanteplusreports.number_of_confirmed_cases_for_severe_malaria_and_put_on_treatm
...
+isanteplusreports.pnls.result.35-39.label=35-39 Yrs
+isanteplusreports.pnls.result.40-44.label=40-44 Yrs
+isanteplusreports.pnls.result.45-49.label=45-49 Yrs
+isanteplusreports.pnls.result.50.label=50+ Yrs
+isanteplusreports.pnls.result.UnKnown.label=Inconnue Yrs
+isanteplusreports.pnls.result.Total.label=Total
+isanteplusreports.pnls.indicator.adult.label=Indicateurs du rapport mensuel ART
+isanteplusreports.pnls.result.pdf.header.label=Rapport mensuel ART
+isanteplusreports.pnls.result.pdf.creationDate.label=date cr��e
+isanteplusreports.newlyEnrolledPatientsOnArt=NOUVEAU 1.1: Nombre de s�ropositifs nouvellement inscrits au TAR pendant la p�riode s�lectionn�e (TX_NEW)
```
Comment (moshonk) [inline api/src/main/resources/messages_fr.properties:263]: I think the translation should be *"Nombre de personnes VIH+ nouvellement enrolées  sous ARV au cours du mois (TX_NEW)"* . Remove  *"NOUVEAU 1.1"*

File: omod/src/main/java/org/openmrs/module/isanteplusreports/conventer/StringToPnlsIndicatorListConverter.java (hunk 3 above)
Comment (mozzy11) [reply omod/src/main/java/org/openmrs/module/isanteplusreports/conventer/StringToPnlsIndicatorListConverter.java:32]: this class is a spring bean that acts as a converter in the controller, to convert the string parameter indicator List to the List<PnlsReportSelectedIndicator>
here
https://github.com/IsantePlus/openmrs-module-isanteplusreports/blob/84da7740cd838429ddcd1e3f185c8a87f1a9e2a3/omod/src/main/java/org/openmrs/module/isanteplusreports/page/controller/PnlsReportPageController.java#L51

File: api/src/main/resources/messages.properties [hunk 5]
```diff
@@ -245,4 +245,32 @@ isanteplusreports.number_of_suspected_cases_of_malaria_tested=Number of suspecte
...
+isanteplusreports.pnls.result.45-49.label=45-49 Yrs
+isanteplusreports.pnls.result.50.label=50+ Yrs
+isanteplusreports.pnls.result.UnKnown.label=UnKnown Yrs
+isanteplusreports.pnls.result.Total.label=Total
+isanteplusreports.pnls.indicator.adult.label=ART Monthly Report Indicators
+isanteplusreports.pnls.result.pdf.header.label=ART Monthly Report
+isanteplusreports.pnls.result.pdf.creationDate.label=Date Created
+isanteplusreports.newlyEnrolledPatientsOnArt=NEW 1.1: Number of HIV+ newly enrolled on ART during the selected period (TX_NEW)
+isanteplusreports.referredInPatientsAnArt=REFERENCE previously enrolled People on ARVs and referees during the month
+isanteplusreports.newlyEnrolledBreastFeedingWomenOnArt=New breastfeeding women enrolled on ARVs
```
Comment (moshonk) [inline api/src/main/resources/messages.properties:276]: For ease of validation and consistency, please include the indicator number before the translation.

File: api/src/main/resources/messages.properties [hunk 6]
```diff
@@ -245,4 +245,32 @@ isanteplusreports.number_of_suspected_cases_of_malaria_tested=Number of suspecte
...
+isanteplusreports.pnls.result.40-44.label=40-44 Yrs
+isanteplusreports.pnls.result.45-49.label=45-49 Yrs
+isanteplusreports.pnls.result.50.label=50+ Yrs
+isanteplusreports.pnls.result.UnKnown.label=UnKnown Yrs
+isanteplusreports.pnls.result.Total.label=Total
+isanteplusreports.pnls.indicator.adult.label=ART Monthly Report Indicators
+isanteplusreports.pnls.result.pdf.header.label=ART Monthly Report
+isanteplusreports.pnls.result.pdf.creationDate.label=Date Created
+isanteplusreports.newlyEnrolledPatientsOnArt=NEW 1.1: Number of HIV+ newly enrolled on ART during the selected period (TX_NEW)
+isanteplusreports.referredInPatientsAnArt=REFERENCE previously enrolled People on ARVs and referees during the month
```
Comment (moshonk) [inline api/src/main/resources/messages.properties:275]: This translation doesn't make sense. I think it should read, 

>"People previously enrolled on ARVs and transferred-in during the month"

File: api/src/main/resources/messages_fr.properties [hunk 7]
```diff
@@ -236,3 +236,31 @@ isanteplusreports.number_of_confirmed_cases_for_severe_malaria_and_put_on_treatm
...
+isanteplusreports.referredInPatientsAnArt=R�F�RENCE a d�j� inscrit des personnes aux ARV et des arbitres au cours du mois
+isanteplusreports.newlyEnrolledBreastFeedingWomenOnArt=Nouvelles femmes allaitantes inscrites aux ARV
```
Comment (moshonk) [inline api/src/main/resources/messages_fr.properties:265]: For ease of validation and consistency, please include the indicator number before the translation.

Comment (moshonk) [reply api/src/main/resources/messages_fr.properties:265]: Also, for French translations, please use what is contained in the "Lexique ARV" sheet

File: api/src/main/resources/messages_fr.properties [hunk 8]
```diff
@@ -236,3 +236,31 @@ isanteplusreports.number_of_confirmed_cases_for_severe_malaria_and_put_on_treatm
...
+isanteplusreports.pnls.result.40-44.label=40-44 Yrs
+isanteplusreports.pnls.result.45-49.label=45-49 Yrs
+isanteplusreports.pnls.result.50.label=50+ Yrs
+isanteplusreports.pnls.result.UnKnown.label=Inconnue Yrs
+isanteplusreports.pnls.result.Total.label=Total
+isanteplusreports.pnls.indicator.adult.label=Indicateurs du rapport mensuel ART
+isanteplusreports.pnls.result.pdf.header.label=Rapport mensuel ART
+isanteplusreports.pnls.result.pdf.creationDate.label=date cr��e
+isanteplusreports.newlyEnrolledPatientsOnArt=NOUVEAU 1.1: Nombre de s�ropositifs nouvellement inscrits au TAR pendant la p�riode s�lectionn�e (TX_NEW)
+isanteplusreports.referredInPatientsAnArt=R�F�RENCE a d�j� inscrit des personnes aux ARV et des arbitres au cours du mois
```
Comment (moshonk) [inline api/src/main/resources/messages_fr.properties:264]: Use what is contained in the "Lexique ARV" sheet

File: api/src/main/resources/messages_fr.properties (hunk 7 above)
Comment (moshonk) [inline api/src/main/resources/messages_fr.properties:265]: please use what is contained in the "Lexique ARV" sheet

Comment (mozzy11) [reply api/src/main/resources/messages_fr.properties:265]: oh cool ,

**Pull request merged**